native/src/modules/endpoint/sagas/fetchResourceCache.js

```diff
@@ -19,6 +19,29 @@ const createErrorMessage = (fetchResult: FetchResultType) => {
     `${message}'Failed to download ${result.url} to ${path}': ${result.errorMessage}\n`, '')
 }
 
+function * watchOnProgress (channel: Channel<number>): Saga<void> {
+  let prevStep = 0
+  while (prevStep < 1) {
+    const progress = yield take(channel)
+    const stepWidth = 10
+    const newStep = Math.floor(progress * stepWidth) / stepWidth
```

Comment:
Couldn't this same behavior achieved by rounding to one decimal place?

Reply:
Yes, it could - but plain javascript does not support truncation to a given number of decimal places